resources/providers/gcplib/inventory/provider.go

```diff
@@ -189,10 +258,109 @@ func mergeAssetContentType(assets []*assetpb.Asset) []*assetpb.Asset {
 			item.IamPolicy = asset.IamPolicy
 		}
 	}
-
 	var results []*assetpb.Asset
 	for _, asset := range resultsMap {
 		results = append(results, asset)
 	}
 	return results
 }
+
+// extends the assets with the project and organization display name
+func extendAssets(ctx context.Context, crm *ResourceManagerWrapper, assets []*assetpb.Asset) []*ExtendedGcpAsset {
+	var extendedAssets []*ExtendedGcpAsset
+	var cache = make(map[string]*fetching.EcsGcpCloud)
+	for _, asset := range assets {
+		keys := getAssetIds(asset)
+		if ecsGcpCloudCached, ok := cache[keys.cacheKey]; ok {
+			extendedAssets = append(extendedAssets, &ExtendedGcpAsset{
+				Asset:       asset,
+				EcsGcpCloud: ecsGcpCloudCached,
+			})
+			continue
+		}
+		cache[keys.cacheKey] = getEcsGcpCloudData(ctx, crm, keys)
+		extendedAssets = append(extendedAssets, &ExtendedGcpAsset{
+			Asset:       asset,
+			EcsGcpCloud: cache[keys.cacheKey],
+		})
+	}
+	return extendedAssets
+}
+
+type GcpAssetIDs struct {
+	orgId         string
+	projectId     string
+	parentProject string
+	parentOrg     string
+	cacheKey      string
+}
+
+func getAssetIds(asset *assetpb.Asset) GcpAssetIDs {
+	orgId := getOrganizationId(asset.Ancestors)
+	projectId := getProjectId(asset.Ancestors)
+	parentProject := fmt.Sprintf("projects/%s", projectId)
+	parentOrg := fmt.Sprintf("organizations/%s", orgId)
+	return GcpAssetIDs{
+		orgId:         orgId,
+		projectId:     projectId,
+		parentProject: parentProject,
+		parentOrg:     parentOrg,
+		cacheKey:      fmt.Sprintf("%s/%s", parentProject, parentOrg),
+	}
+}
+
+func getEcsGcpCloudData(ctx context.Context, crm *ResourceManagerWrapper, keys GcpAssetIDs) *fetching.EcsGcpCloud {
+	var orgName string
+	var projectName string
+	wg := sync.WaitGroup{}
+	wg.Add(1)
+	go func() {
+		orgName = crm.getOrganizationDisplayName(ctx, keys.parentOrg)
+		wg.Done()
+	}()
+	wg.Add(1)
+	go func() {
+		projectName = crm.getProjectDisplayName(ctx, keys.parentProject)
+		wg.Done()
+	}()
+	wg.Wait()
+	return &fetching.EcsGcpCloud{
+		ProjectId:        keys.projectId,
+		ProjectName:      projectName,
+		OrganizationId:   keys.orgId,
+		OrganizationName: orgName,
+	}
+}
+
+func getOrganizationId(ancestors []string) string {
+	last := ancestors[len(ancestors)-1]
```

Comment:
what happens if we asses a project that is not associated with an organization?

Reply:
are you sure this is a real secnario ? i don't think users can do that. the `Ancestors` field description on the asset says an organization is always the last item. 

technically, what will happen here if it is a real case is that the project id would be the same as the org id, as both would access the same member

Reply:
we can add custom logic to omit the `orgId` and `orgName` if `len(ancestors)` is `1`. but i think the juice isn't worth the squeeze, because:
1. we can't use struct tags, so this means sprinkling a bunch of if statements which kinda pollutes the code for this edge case.
2. this scenario is unlikely to happen for companies, as those usually use organizations to scope projects. although, GCP does not require an organization.
3. in the case where there is no organization, IMO it's ok the return the org as the project, hardly an issue. 

wdyt?